### PR TITLE
use original file/directory with local pyflyte run

### PR DESCRIPTION
**Pull request by cosmicBboy**
# TL;DR

Fixes #4426. This PR makes it so that local executions use the original file/directory path instead of creating a temporary copy.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - ~~[ ] Code documentation added~~
 - ~~[ ] Any pending items have an associated Issue~~

## Complete description

This PR makes modifications to the underlying click types and FlyteDirectory type so that the original files and directories are used.

## Tracking Issue
[https://github.com/flyteorg/flyte/issues/<number>](https://github.com/flyteorg/flyte/issues/4426)

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/interaction/click_types.py [hunk 1]
```diff
@@ -57,8 +57,11 @@ def convert(
         self, value: typing.Any, param: typing.Optional[click.Parameter], ctx: typing.Optional[click.Context]
     ) -> typing.Any:
         p = pathlib.Path(value)
+        # set remote_directory to false if running pyflyte run locally. This makes sure that the original
+        # directory is used and not a random one.
+        remote_directory = None if getattr(ctx.obj, "is_remote", None) else False
```
Comment (kumare3) [inline flytekit/interaction/click_types.py:62]: i just dont like this false thing, but what can you do.
I wanted to add FlyteFile.for_remote() or something like that

**Review by kumare3 — APPROVED** [review]

**Review by kumare3 — APPROVED** [review]
Check why the spark tests are broken

**Comment by cosmicBboy** [comment]
@kumare3 can you re-+1? the pyspark tests are unrelated, they're fixed by this: https://github.com/flyteorg/flytekit/pull/2007

**Review by eapolinario — APPROVED** [review]
